addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2668,18 +2668,118 @@ options.registry.CookiesBar = options.registry.SnippetPopup.extend({
     },
 });
 
+/**
+ * Base class for all cover updaters.
+ */
+const CoverUpdaterMixin = (parentClass) => {
+    const Extended = parentClass.extend({
+        /**
+         * @constructor
+         */
+        init() {
+            this._super.apply(this, arguments);
+    
+            this.$image = this.$target.closest('.o_record_cover_container').find('.o_record_cover_image');
+        },
+    
+        //--------------------------------------------------------------------------
+        // Options
+        //--------------------------------------------------------------------------
+    
+        /**
+         * Called to mark cover-specific dirty for options that are not cover-only.
+         * @see this.selectClass for parameters
+         */
+        updateCover(previewMode, widgetValue, params) {
+            if (!previewMode) {
+                this._updateSavingDataset();
+            }
+        },
+    
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+    
+        /**
+         * Adds the update cover request to all inputs.
+         *
+         * @override
+         */
+        async _renderCustomXML(uiFragment) {
+            await this._super(...arguments);
+            uiFragment.querySelectorAll('we-select, we-button, we-input, we-colorpicker, we-imagepicker').forEach(el => {
+                el.dataset.updateCover = true;
+            });
+        },
```

Comment:
Meh... why not ^^ But maybe this would be a bit cleaner to override `_select` directly. We could add a new `onOptionUsed` or something like that after the options have been called:

```javascript
        // Call each option method sequentially
        for (const methodName of widget.getMethodsNames()) {
            const widgetValue = widget.getValue(methodName);
            const params = widget.getMethodsParams(methodName);

            if (params.applyTo) {
                if (!$applyTo) {
                    $applyTo = this.$(params.applyTo);
                }
                const proms = _.map($applyTo, subTargetEl => {
                    const proxy = createPropertyProxy(this, '$target', $(subTargetEl));
                    return this[methodName].call(proxy, previewMode, widgetValue, params);
                });
                await Promise.all(proms);
            } else {
                await this[methodName](previewMode, widgetValue, params);
            }
        }

        this.onOptionUsed(previewMode);
```
Then in this specific mixin, you can implement it.